src/main/java/com/aws/greengrass/clientdevices/auth/certificate/ClientCertificateGenerator.java

```diff
@@ -77,9 +77,7 @@ public synchronized void generateCertificate(Supplier<List<String>> connectivity
                     .kv("certExpiry", getExpiryTime())
                     .log("New client certificate generated");
 
-            X509Certificate caCertificate = certificateStore.getCACertificate();
-            X509Certificate[] chain = {certificate, caCertificate};
-            callback.accept(chain);
+            callback.accept(certificate, certificateStore.getCaCertificateChain());
```

Comment:
I still don't love that things in the certificate store can change midway through this flow. Fixing that will require bigger changes in the certificate store to essentially get a "snapshot".

Let's not try to fix that here, but just keep it in mind. (BTW, I am not expecting customers to change the CA type from managed to provided as doing so requires them to have access to the device to install keys)